client/src/ui/organisms/article-actions/language-menu/index.tsx

```diff
@@ -127,3 +126,37 @@ function LanguageMenuItem({
     </a>
   );
 }
+
+function LocaleRedirectSetting() {
+  const gleanClick = useGleanClick();
+  const locale = useLocale();
+  const [value, setValue] = useState(false);
+
+  useEffect(() => {
+    setValue(!!getCookieValue(PREFERRED_LOCALE_COOKIE_NAME));
+  }, [setValue]);
+
+  function toggle(event) {
+    const newValue = event.target.checked;
+    if (newValue) {
+      if (!getCookieValue(PREFERRED_LOCALE_COOKIE_NAME)) {
+        setCookieValue(PREFERRED_LOCALE_COOKIE_NAME, locale, {
+          maxAge: 60 * 60 * 24 * 365 * 3,
```

Comment:
3 years?
Perhaps define a constant, or add a comment

Reply:
Fixed in 2f348f7e58.